### PR TITLE
Fix #114 (Unified multiple value reponse)

### LINKED ISSUE #114: Unified multiple value reponse
### Setup

According to old tests : if the specified id is not found, the operation must send a 404 error whereas this error is not sent anymore.
Indeed, if the operation returns an array, it fills the response by as much `null` as not found ids and sends a 200 success.
The problem is that clients don't know if an error occurred or not.
The other problematic is "_do we have to continue the process if an error occurred ?_"
### Solution proposed

A good way to fix it could be processing over all elements of the array parameter and send : 
- As much `null` as errors
- 200 (or 201) if all succeeded
- 207 if some succeeded and some didn't
- 404 if no ids were found at all

### PR DESCRIPTION
**Pull request by zankia**

### CODE REVIEW

**Comment by gnoxr** [comment]
### Review of the changes

My thoughts on it.

- `damas.js` :+1: LGTM (looks good to me).

- `db/mongodb.js` :hand: 
> :+1: Everything looking like a valid ObjectID in `create()` is converted to an ObjectID, thus solving the major problems implied by using custom IDs.
> :+1: The functions are uniformized to return valid arrays. It might be improved a bit though. **If it was tested** (by running the unit tests), then it is ok.
> :-1: The `remove()` method returns booleans, but we need it to return the objects which were removed, or at least the ids. It could be changed by pushing `ids_o[cursor]` instead of `result.result.n !== 0`.

- `server-tests/tests-frisby_spec.js` :+1: LGTM
> I don't have much to say, there are new tests and some existing ones were updated. This does not affect the existing system.

- `routes/cruds.js` :+1: LGTM
> The main change is the use of the `getMultipleResponse()` function, which parses the output of a DB method to determine what are the most appropriate statuses. **However**, this function might be improved a little, but is currently good. **If it was tested** (by running the unit tests), then it is ok.

**Comment by gnoxr** [comment]
With this last commit, everything LGTM :smiley:

File: server-nodejs/routes/cruds.js [hunk 1]
```diff
@@ -453,6 +523,34 @@ module.exports = function (app, express) {
     }; // getFile()
 
 
+    /**
+     * Sets the appropriate response and status code for multiple params or not
+     * @param {boolean} isArray - did client sent an array or not
+     * @param {array} doc - the database response
+     * @param {} result - the returned object or string
```
Comment (remyla) [inline server-nodejs/routes/cruds.js]: the inline doc could be updated

**Comment by remyla** [comment]
thank you

**Pull request merged**